### PR TITLE
Add support for recursively looking at directories when uploading imagesets

### PR DESCRIPTION
**Pull request by exhaustedbrit**

### CODE REVIEW

**Comment by exhaustedbrit** [comment]
> this behaviour needs to be opt-in based on a command line argument

I saw that in the ticket after moving it, I think so too - I'll hide it behind a flag.

File: zeg/imagesets.py [hunk 1]
```diff
@@ -338,21 +338,34 @@ def delete(log, session, args):
...
-                entry.path for entry in os.scandir(path)
-                if entry.is_file() and entry.name.lower().endswith(allowed_ext)
-            )
+            if should_recursive:
+                resolved.extend(
+                    _scan_directory_tree(path, allowed_ext)
+                )
+            else:
+                resolved.extend(
+                    entry.path for entry in os.scandir(path)
```
Comment (ahaith) [inline zeg/imagesets.py:354]: Does the list comprehension not need to be wrapped in `[]`?

Comment (ahaith) [reply zeg/imagesets.py:354]: It seems not. Interesting.

**Comment by exhaustedbrit** [comment]
Just tweaked it slightly to use the configuration object, as the forced mime type will work that way too but don't want it as part of this PR.

**Comment by ahaith** [comment]
> Just tweaked it slightly to use the configuration object, as the forced mime type will work that way too but don't want it as part of this PR.

Cool, I agree that's neater.

File: zeg/__main__.py [hunk 2]
```diff
@@ -181,6 +181,12 @@ def _add_standard_args(parser):
         action='store_true',
         help='Enable verbose logging.',
     )
+    parser.add_argument(
```
Comment (ahaith) [inline zeg/__main__.py:184]: Can remove this now then

Comment (exhaustedbrit) [reply zeg/__main__.py:184]: Actually, I was assigning to the config object via args and hadn't intended for it to be specified in the yaml, that might be better, I'll raise the PR for mime types and see what you think, it's branched off of this

Comment (exhaustedbrit) [reply zeg/__main__.py:184]: Could just have both?

Comment (exhaustedbrit) [reply zeg/__main__.py:184]: Have made it so you can specify either via a CLI arg or as the configuration object, the configuration yaml takes precedence

Comment (ahaith) [reply zeg/__main__.py:184]: Possibly. Although i feel it's much more intuitive to have it next to where the paths are defined.

Comment (exhaustedbrit) [reply zeg/__main__.py:184]: Will remove the CLI arguments then

**Comment by exhaustedbrit** [comment]
The other PR has the readme and documentation for the mime type feature, so merging this now.

**Pull request merged**